packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -221,6 +240,18 @@ export interface ClusterAttributes {
    */
   readonly kubectlPrivateSubnetIds?: string[];
 
+  /**
+   * An Open ID Connect issuer url that was configured for the cluster
+   * @default - if no value specified you won't be able to add an IRSA in your cluster
+   */
+  readonly openIdConnectIssuerUrl?: string;
+
+  /**
+   * An Imported Open ID Connect provider for this cluster
+   * @default - if no value specified - cdk will create new provider when you add IRSA
+   */
+  readonly openIdConnectProvider?: iam.IOpenIdConnectProvider;
```

Comment:
Does it make sense to configure both the provider and the url? what if they have different values? Isn't configuring the issuer url sufficient?

Reply:
this field openIdConnectProvider is for imported, already existing provider 
but if you don't have provider configured and you want cdk to create one for you - you can put only URL and that's it

Reply:
if you provide openIdConnectProvider, the URL will be set from the configuration of that provider

Reply:
so you need one of 

url - to create the new provider
provider - to set existing provider

Reply:
So we should validate that only one of them is passed. Something still doesn't sit well with this modeling. I wonder if we should just accept an `IOpenIdConnectProvider` and thats it. We can then move this logic:

https://github.com/aws/aws-cdk/blob/aebac9261dfaa37795a91ddced164bb6a0841f13/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L1238-L1250

Into a specific `eks.OpenIdConnectProvider` construct that implements `IOpenIdConnectProvider` so that users are able to easily create compatible providers if they don't already have one.

```ts
// existing provider
const cluster = eks.Cluster.fromClusterAttributes(this, 'Cluster', {
  openIdConnectProvider: iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(...), 
});

// create provider 
const cluster = eks.Cluster.fromClusterAttributes(this, 'Cluster', {
  openIdConnectProvider: new eks.OpenIdConnectProvider(this, 'Provider', issuerUrl), 
});
```

@eladb Can I get you take here?

Reply:
yeah I think it would be better 
because in this case, we will force user to explicitly create a provider instead of implicit creation. I will push a change

Reply:
ah I think we need to remain the behaviour as is in the new Cluster
but apply this to the ImportedCluster

Reply:
Yes, when creating a cluster the provider creation is implicit 👍

Reply:
@eladb You ok with this approach?

Reply:
@iliapolo check this one https://github.com/aws/aws-cdk/pull/10774/commits/05851862a602751338f7ef754ee162ad14fbc6de